synapse/storage/schema/main/delta/74/03thread_notifications_not_null.sql.sqlite

```diff
@@ -0,0 +1,108 @@
+/* Copyright 2022 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```

Comment:
One is that we haven't played with foreign key constraints here. I'm guessing we don't need to — doesn't seem likely that other tables exist with constraints on this table — but wouldn't hurt to say this aloud.

Reply:
>  You're mostly there but a small divergence.

Is it just the foreign key constraint or is there something else missing?

Reply:
(And to confirm -- nothing has foreign keys to this table!)

Reply:
Thanks for sharing! I think the steps I did are copied from other schema deltas, seems like dropping the indexes is completely unneeded since dropping the table will do it. And then moving those after the insertions is the main thing, as you said.

Reply:
yup that's all. I think I saw something else but then realised I hallucinated it